api/v1beta2/foundationdbcluster_types.go

```diff
@@ -1994,3 +1994,39 @@ func (cluster *FoundationDBCluster) GetUseUnifiedImage() bool {
 func (cluster *FoundationDBCluster) GetIgnoreTerminatingPodsSeconds() int {
 	return pointer.IntDeref(cluster.Spec.AutomationOptions.IgnoreTerminatingPodsSeconds, int((10 * time.Minute).Seconds()))
 }
+
+// AddProcessGroupsToRemovalList adds the provided process group IDs to the remove list.
+// If a process group ID is already present on that list it won't be added a second time.
+func (cluster *FoundationDBCluster) AddProcessGroupsToRemovalList(processGroupIDs []string) {
+	removals := map[string]None{}
+
+	for _, id := range cluster.Spec.ProcessGroupsToRemove {
```

Comment:
In theory would could check if a process is already on the `ProcessGroupsToRemoveWithoutExclusion` and if so skip it, because if the process group is already on the `ProcessGroupsToRemoveWithoutExclusion` it will be removed anyway without waiting for the exclusion. I didn't implemented this because I think the benefit will not be huge.